src/model/formats/foreflight-utils.ts

```diff
@@ -22,6 +28,34 @@ export class ForeFlightUtils {
   static readonly encoder = new TextEncoder();
   static readonly decoder = new TextDecoder();
 
+  public static readonly CHECKLIST_ITEM_PREFIXES = new Map([
+    [ChecklistItem_Type.ITEM_PLAINTEXT, ''],
+    [ChecklistItem_Type.ITEM_NOTE, 'NOTE: '],
+    [ChecklistItem_Type.ITEM_CAUTION, 'CAUTION: '],
+    [ChecklistItem_Type.ITEM_WARNING, 'WARNING: '],
+  ]);
+
+  public static readonly CHECKLIST_ITEM_TYPES = new Map(
+    [...ForeFlightUtils.CHECKLIST_ITEM_PREFIXES.entries()].map(
+      (item) => item.reverse() as [string, ChecklistItem_Type],
+    ),
+  );
+
+  public static promptToPartialChecklistItem(prompt: string): PartialChecklistItem {
+    const matches = /^([^:]+: )?(.*)$/.exec(prompt || '');
```

Comment:
Do you need the first group to be consumed non-greedly?
i.e. "Foo: bar: baz" should probably get a prefix of "Foo: ", but with this it gets "Foo: bar: "

Reply:
> "Foo: bar: baz" should probably get a prefix of "Foo: ", but with this it gets "Foo: bar: "

Are you sure about this? This is exactly what I was trying to avoid with my `[^:]` hat trick. According to the regex tester and my unit test, it seems to work.

<img width="1189" alt="Screenshot 2024-08-26 at 11 38 22" src="https://github.com/user-attachments/assets/61050ac6-d8dd-45e8-84d2-e1840f514ab7">

Reply:
And that's what happens when you review code late at night :) You're right, sorry about that.

Reply:
Late night or not, sometimes one can overlook something... but always better to ask, could have been a real problem, so thank you anyway!